src/Intersection.ts

```diff
@@ -81,6 +81,36 @@ export class Intersection {
     }
   }
 
+  /**
+   * Use the ray casting algorithm to determine if {@link point} is in the polygon defined by {@link points}
+   * @see https://en.wikipedia.org/wiki/Point_in_polygon
+   * @param point
+   * @param points polygon points
+   * @returns
+   */
+  static isPointInPolygon(point: Point, points: Point[]) {
+    const other = new Point(point).setX(
+      Math.min(point.x - 1, ...points.map((p) => p.x))
+    );
+    let hits = 0;
+    for (let index = 0; index < points.length; index++) {
+      const inter = this.intersectSegmentSegment(
+        // polygon side
+        points[index],
+        points[(index + 1) % points.length],
+        // ray
+        point,
+        other
+      );
+      if (inter.includes(point)) {
+        // point is on the polygon side
+        return true;
+      }
```

Comment:
is this an optimization or is needed to cover extra use cases?
This would help if the polygon had a tons of points and we are checking for containing of one of the early points in the list. removing it on a 4 point polygon removes another 20% of speed loss

Reply:
If the ray is casted in this case it will return an even number of points and fail the ray casting test

Reply:
but you can concat all the results and check at the end if that is better

Reply:
I bet if we make the array of points a Set it will help also

Reply:
I think we should move to Set wereever possible

Reply:
if having the point exactly over the segment does not count as intersection we can leave this code as it is. I wanted to be sure it wasn't a speed optimization of some sort. It should be written in a comment that being on the segment doesn't count as intersecting, so we don't forget.